r/R/make_pseudobulk_matrix.R

```diff
@@ -0,0 +1,62 @@
+#' Make aggregated matrix for pseudo bulk differential expression
+#'
+#' The SeuratObject requires a `custom` meta.data slot, specifying which cells
+#' belong to the `base` and `background` groups. It filters out cells with `NA` in `custom`.
+#' Also, a `samples` meta.data slot is required, specifying samples.
+#'
+#' @param scdata a SeuratObject,
+#'
+#' @return a SeuratObject with counts aggregated by sample for one cluster
+#' @export
+#'
+makePseudobulkMatrix <- function(scdata) {
+  # filter out cells not in base/background groups
+  scdata <- scdata[, !is.na(scdata@meta.data$custom)]
+
+  counts <- scdata[["RNA"]]@counts
+  gene_annotations <- scdata@misc$gene_annotations
+
+  groups <- factor(scdata@meta.data$samples)
```

Comment:
```suggestion
  groups <- factor(scdata$samples)
```

Reply:
yep, explicit is better. It was easier to type when playing around, then it stuck!